lib/plugin.js

```diff
@@ -71,9 +72,15 @@ ManifestPlugin.prototype.apply = function(compiler) {
       return memo;
     }, {}));
 
-    // Append optional basepath onto all references.
-    // This allows output path to be reflected in the manifest.
-    if (this.opts.basePath) {
+    if (this.opts.useBasePathAndPublicPath) {
```

Comment:
I agree, it *is* clunky - but I was trying to keep backwards compatibility. It looks like we should potentially just solve this in v2 properly